code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/config/GithubConfigManager.java

```diff
@@ -0,0 +1,208 @@
+            }
+            createOrUpdateSetting("api.auth.github.allow.users", toCommaSeparatedString(userList));
+        }
+        if (null != config.get("allowOrganizations")) {
+            List<String> orgList;
+            try {
+                orgList = appendOrgIds((List<String>) config.get("allowOrganizations"));
+            } catch (ClassCastException e) {
+                throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "allowedOrganizations should be a list");
+            }
+            createOrUpdateSetting("api.auth.github.allow.orgs", toCommaSeparatedString(orgList));
+        }
+
+        return new Object();
+    }
+
+    protected List<String> appendUserIds(List<String> usernames) {
+        if(usernames == null) {
+            return null;
+        }
+        return Lists.transform(usernames, new Function<String, String> () {
+
+            @Override
+            public String apply(String username) {
+                GithubAccountInfo userInfo = client.getUserIdByName(username);
+                if(userInfo == null) {
+                    throw new RuntimeException("Invalid username: " + username);
+                }
+                return userInfo.toString();
+            }
+            
+        });
+    }
+    
+    protected List<String> appendOrgIds(List<String> orgs) {
+        if(orgs == null) {
+            return null;
+        }
+        return Lists.transform(orgs, new Function<String, String> () {
+
+            @Override
+            public String apply(String org) {
+                GithubAccountInfo orgInfo = client.getOrgIdByName(org);
+                if(orgInfo == null) {
+                    throw new RuntimeException("Invalid organization: " + org);
+                }
+                return orgInfo.toString();
+            }
+            
+        });
+    }
+    
+    private void createOrUpdateSetting(String name, Object value) {
+        Setting setting = objectManager.findOne(Setting.class, "name", name);
+        if (null == setting) {
+            objectManager.create(Setting.class, "name", name, "value", value);
+        } else {
+            objectManager.setFields(setting, "value", value);
+        }
+        DeferredUtils.defer(new Runnable() {
+            
+            @Override
+            public void run() {
+                ArchaiusUtil.refresh();
+            }
+        });
+    }
+
+    @Override
+    protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
+        boolean enabled = SECURITY.get();
+        String clientId = GITHUB_CLIENT_ID.get();
+        List<String> allowedUsers = getAccountNames(fromCommaSeparatedString(GITHUB_ALLOWED_USERS.get()));
+        List<String> allowedOrgs = getAccountNames(fromCommaSeparatedString(GITHUB_ALLOWED_ORGS.get()));
+        return new GithubConfig(enabled, clientId, allowedUsers, allowedOrgs);
```

Comment:
List makes no sense, `listInternal()` should return a list, not a single object.  Was this what @vincent99 was complaining about?